cypress/integration/tag_amd_spec.js

```diff
@@ -0,0 +1,610 @@
+                cy.get('#tags').type('{backspace}{backspace}'); // remove third
+                cy.get('#tags').should('have.attr', 'maxlength').and('eq', '20');
+            });
+
+            it('should not update maxlength after removing a tag if maxlength is not present', () => {
+                cy.get('#tags').should('not.have.attr', 'maxlength');
+                cy.get('#tags').type('tag{enter}');
+                cy.get('#tags').type('{backspace}{backspace}');
+                cy.get('#tags').should('not.have.attr', 'maxlength');
+            });
+
+            it('should update maxlength after removing last tag if maxlength is present', () => {
+                cy.get('#tags').invoke('attr', 'maxlength', '200');
+                cy.get('#tags').type('tag{enter}');
+                cy.get('#tags').type('{backspace}{backspace}');
+                cy.get('#tags').should('have.attr', 'maxlength').and('eq', '200');
+            });
+
+            it('should update maxlength after removing not the last tag if maxlength is present', () => {
+                var tag1 = 'tag1',
+                    tag2 = 'tag2',
+                    separator = ',',
+                    initialMaxlength = 200,
+                    intermediateMaxlength = initialMaxlength - tag1.length - separator.length - tag2.length - separator.length,
+                    finalMaxlength = initialMaxlength - tag1.length - separator.length;
+
+                cy.get('#tags').invoke('attr', 'maxlength', `${initialMaxlength}`);
+                cy.get('#tags').type(`${tag1}${separator}${tag2}{enter}`);
+                cy.get('#tags').should('have.attr', 'maxlength').and('eq', `${intermediateMaxlength}`);
+                cy.get('#tags').type('{backspace}{backspace}');
+                cy.get('#tags').should('have.attr', 'maxlength').and('eq', `${finalMaxlength}`);
+            });
+        });
+    });
+
+    context('Callbacks', () => {
+        let tag = null,
+            errorCalled = false,
+            callback = {},
+            win = null
+
+        beforeEach(() => {
+            cy.visit('/?test=callbacks');
+            cy.window().then((w) => win = w);
+        });
+
+        afterEach(() => {
+            tag = null,
+            errorCalled = false,
+            callback = {},
+            win = null;
+        });
+
+        it('should call errorCleared callback with input as argument when removing error class', () => {
+            callback.errorCleared = (param) => {
+                expect(errorCalled).to.be.false;
+                errorCalled = true;
+                expect(errorCalled).to.be.true;
+                expect(callback.errorCleared).to.always.be.calledWith(param);
+            };
+            cy.spy(callback, 'errorCleared');
+
+            cy.get('#tags').type('tag{enter}');
+            win.define(['doc', 'tag'], ($, t) => tag = t);
+
+            cy.spy(tag, 'tagify');
```

Comment:
Could this line and the line above both be in `beforeEach`?

Reply:
I tried change this but don't worked. :(